game.h

```diff
@@ -127,7 +117,7 @@ class game
   int get_dist_x_pls() const noexcept { return m_dist_x_pls; }
 
   ///Manages collisons with walls
-  player wall_collision(player p);
+  player resolve_wall_collision(player p);
```

Comment:
From the function prototype, I find it hard to guess what it does: it takes a player, modifies it and puts it back in game I guess? This is one case where `void resolve_wall_collision(player& p);` would be better. Also, why is it not a const member function (as I guess the arena does not change?).

Still, this has nothing to do with the PR, so I will accept :-)